client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -121,10 +139,10 @@ public String getServiceAuthority() {
         @Override
         @Synchronized
         public void start(Listener listener) {
-            Preconditions.checkState(started.compareAndSet(false, true));
+            Preconditions.checkState(!started.get());
```

Comment:
It would be better if we either make this No Op (if its already started) Or make this method idempotent. 
This is anyway synchronized, and it appears that the contract is that this method can be called multiple times.